pkg/build/builder/cmd/scmauth/util.go

```diff
@@ -7,19 +7,43 @@ import (
 	"path/filepath"
 	"strings"
 
-	builder "github.com/openshift/builder/pkg/build/builder"
+	buildv1 "github.com/openshift/api/build/v1"
+	"github.com/openshift/builder/pkg/build/builder"
 	utillog "github.com/openshift/builder/pkg/build/builder/util/log"
 )
 
 var log = utillog.ToFile(os.Stderr, 2)
 
-func createGitConfig(includePath string, context SCMAuthContext) error {
+func createGitConfig(includePath string, context SCMAuthContext, gitSource *buildv1.GitBuildSource) error {
 	tempDir, err := ioutil.TempDir("", "git")
 	if err != nil {
 		return err
 	}
 	gitconfig := filepath.Join(tempDir, ".gitconfig")
-	content := fmt.Sprintf("[include]\npath = %s\n", includePath)
+	content := ""
+	if len(includePath) > 0 {
+		content = fmt.Sprintf("[include]\npath = %s\n", includePath)
+	}
+	if gitSource != nil {
+		if gitSource.HTTPProxy != nil && len(*gitSource.HTTPProxy) > 0 {
+			content = content + fmt.Sprintf("[http]\nproxy = %s\n", *gitSource.HTTPProxy)
+		}
+		if gitSource.HTTPSProxy != nil && len(*gitSource.HTTPSProxy) > 0 {
+			content = content + fmt.Sprintf("[https]\nproxy = %s\n", *gitSource.HTTPSProxy)
+		}
+		if gitSource.NoProxy != nil && len(*gitSource.NoProxy) > 0 {
+			noProxies := strings.Split(*gitSource.NoProxy, ",")
+			for _, noProxy := range noProxies {
+				if strings.Contains(gitSource.URI, noProxy) {
+					content = content + fmt.Sprintf("[http \"%s\"]\nproxy =\n", gitSource.URI)
+					content = content + fmt.Sprintf("[https \"%s\"]\nproxy =\n", gitSource.URI)
+				}
+			}
+		}
```

Comment:
We could use a comment to explain what we're doing here.

Ex: "To set up NO_PROXY in a .gitconfig file, add `http` and `https` records for the specific URI that unset `proxy`"

Reply:
part of next push